cobbler/utils/__init__.py

```diff
@@ -1354,3 +1354,71 @@ def filelock(lock_file: str):
             with contextlib.suppress(OSError):
                 fcntl.flock(fd, fcntl.LOCK_UN)
             os.close(fd)
+
+
+def merge_dicts_recursive(
+    base_dict: Dict[str, Any], updating_dict: Dict[str, Any], str_append: bool = False
+) -> Dict[str, Any]:
+    """Merge updating_dict into base_config recursively.
+
+    :param base_dict: Base dictionary.
+    :param updating_dict: Updating dict, overrides base_dict.
+    :param str_append: Append the updating_dict String values to the base_dict
+                       with the same key instead of overwriting the original value.
+    :returns dict: Merged dict"""
+    ret = base_dict.copy()
+    for k, v in updating_dict.items():
+        if (
+            k in base_dict
+            and isinstance(v, dict)
+            and isinstance(base_dict.get(k), dict)
+        ):
+            ret[k] = merge_dicts_recursive(base_dict[k], v, str_append)  # type: ignore
+        elif str_append and k in ret and isinstance(v, str):
+            ret[k] += v
+        else:
+            ret[k] = v
+    return ret
+
+
+def create_files_if_not_existing(files: List[str]) -> None:
+    """
+    Create empty files if they don't already exist. If they exist, do nothing.
+
+    :param files: A list of the files to create. If the base directory doesn't
+                  exist, create it too.
+    raises OSError: In case the file cannot be created.
+    """
+    for file in files:
+        if not file:
+            logger.warning("Attempted to create an empty file in %s, skipping" % files)
+            continue
+        if not os.path.exists(file):
+            # It's possible the file dir doesn't exist yet, create it
+            basedir = os.path.dirname(file)
+            if not os.path.exists(basedir):
+                os.makedirs(basedir)
+            # Create empty file
+            open(file, "a").close()
```

Comment:
All of the files we touch are either ASCII or UTF-8. All daemons that we are using understand if their config files are written in UTF-8. As such let's make the linters happy. An ignore comment for the `open` statement would be nice in addition to have the pylint comment to satisfy the linters.

```suggestion
            open(file, "a", encoding="UTF-8").close()  # pylint: disable=consider-using-with
```